pkg/apis/kubedirector/v1beta1/kubedirectorcluster_types.go

```diff
@@ -194,10 +194,11 @@ type BlockStorage struct {
 
 // RoleStatus describes the component objects of a virtual cluster role.
 type RoleStatus struct {
-	Name                string            `json:"id"`
-	StatefulSet         string            `json:"statefulSet"`
-	Members             []MemberStatus    `json:"members"`
-	EncryptedSecretKeys map[string]string `json:"encryptedSecretKeys,omitempty"`
+	Name                string             `json:"id"`
+	StatefulSet         string             `json:"statefulSet"`
+	Members             []MemberStatus     `json:"members"`
+	EncryptedSecretKeys map[string]string  `json:"encryptedSecretKeys,omitempty"`
+	UpgradingMembers    map[string]*string `json:"upgradingMembers,omitempty"`
```

Comment:
We also need to reflect this change in kubedirector.hpe.com_kubedirectorapps_crd.yaml

Reply:
Pardon, `kubedirector.hpe.com_kubedirectorapps_crd.yaml` or `kubedirector.hpe.com_kubedirectorclusters_crd.yaml` ?

Reply:
Actually I updated the `kubedirectorclusters` file, as RoleStatus struct is a part of KDCluster, not KDApp..